src/NUnitEngine/nunit.engine/Services/ExtensionService.cs

```diff
@@ -431,89 +431,99 @@ internal void FindExtensionsInAssembly(ExtensionAssembly assembly)
         {
             log.Info("Scanning {0} assembly for Extensions", assembly.FilePath);
 
-            ValidateTargetFramework(Assembly.GetEntryAssembly(), assembly);
+            if (ValidateTargetFramework(Assembly.GetEntryAssembly(), assembly))
+            {
 
-            IRuntimeFramework assemblyTargetFramework = null;
+                IRuntimeFramework assemblyTargetFramework = null;
 #if !NETSTANDARD2_0
-            var currentFramework = RuntimeFramework.CurrentFramework;
-            assemblyTargetFramework = assembly.TargetFramework;
-            if (!currentFramework.CanLoad(assemblyTargetFramework))
-            {
-                if (!assembly.FromWildCard)
-                {
-                    throw new NUnitEngineException($"Extension {assembly.FilePath} targets {assemblyTargetFramework.DisplayName}, which is not available.");
-                }
-                else
+                var currentFramework = RuntimeFramework.CurrentFramework;
+                assemblyTargetFramework = assembly.TargetFramework;
+                if (!currentFramework.CanLoad(assemblyTargetFramework))
                 {
-                    log.Info($"Assembly {assembly.FilePath} targets {assemblyTargetFramework.DisplayName}, which is not available. Assembly found via wildcard.");
-                    return;
+                    if (!assembly.FromWildCard)
+                    {
+                        throw new NUnitEngineException($"Extension {assembly.FilePath} targets {assemblyTargetFramework.DisplayName}, which is not available.");
+                    }
```

Comment:
According to my more general comment - this if clause is the bit that should be removed. My fault, sorry!